roles/kubernetes/preinstall/templates/resolved.conf.j2

```diff
@@ -10,4 +10,8 @@ Domains={{ ([ 'default.svc.' + dns_domain, 'svc.' + dns_domain ] + searchdomains
 #MulticastDNS=no
 DNSSEC=no
 Cache=no-negative
+{% if ansible_os_family in ["Flatcar", "Flatcar Container Linux by Kinvolk"] %}
```

Comment:
This looks ok to me but it may be a more general solution to deal with distributions running `systemd-resolved`; why limit it to just Flatcar?

Reply:
At least in my environment (Flatcar Linux) this inevitably became a problem.

I've been tracking this issue steadily until very recently, and found that it crashes when the `nodelocaldns' feature is enabled.

I don't think this works the same on other distributions (such as Ubuntu). The `enable_nodelocaldns` feature is already enabled by default, so I know that no other major distributions have reported problems with it.

Another peculiarity is that, at least in Flatcar Linux, the `/etc/resolv.conf` configuration file is not valid, and instead depends only on the `systemd-resolved`. However, there are several `resolvconf` stuffs that are specific to `Flatcar Linux`, but I still don't know why they exist.

Especially,
* https://github.com/kubernetes-sigs/kubespray/blob/master/roles/kubernetes/preinstall/tasks/0060-resolvconf.yml

Reply:
Hmm, ok, lets go ahead with this change then and see if other deployments report this problem, I have a suspicion that ubuntu 22.04 may be affected as well bug since I'm not running that in production anywhere this is just a suspicion.